### DISCUSSION TITLE
Alguém trabalhando com conceito de micro frontends?

### DISCUSSION THREAD

**Discussion by juliovedovatto** [frontendbr/forum#1494] [Discussão]
Mês passado me deparei com este artigo https://martinfowler.com/articles/micro-frontends.html e me interessei pelo conceito, inclusive tem um ótimo material no https://micro-frontends.org

Porém até agora achei mais material teórico do que uma implementação prática. 

Quero montar uma apresentação sobre o assunto e queria trocar figurinhas com alguém que esteja trabalhando com isso.

Lembrando que o conceito é _ligeiramente_ diferente do conceito de components que o os frameworks frontend trazem hoje em dia. Tem gente que já considera React monólito 😄

**Comment by ninetails** [comment]
#540 #661 
plus @houstondapaz

Pessoalmente dei uma estudada sobre, mas achei mais contras que prós e um uso muito condicional. No caso de que depende muito da necessidade, seja trabalhar lado-a-lado com legado ou ter muito recurso e times diferentes pra arquiteturar de um jeito otimizado e organizado.

<sup><sub>**Edit:** removendo crítica pessoal leviana, ambígua e exacerbada.</sub></sup>

**Comment by houstondapaz** [comment]
To numa semana corrida, mas tenho umas experiencias legais (outras nem tanto) para compartilhar, volto aqui depois

**Comment by Juuhz** [comment]
Fala Dev,

Cara, esse conceito de micro-frontend me lembra muito o conceito que o **Atomic Design** traz:
http://bradfrost.com/blog/post/atomic-web-design/

Da uma olhada, pode enriquecer sua apresentação. 

Eu sempre utilizo essa metodologia em meus projetos front-end (ajuda bastante na componentização), caso você queira ver um exemplo prático, eu utilizei o **Atomic Design** em meu teste técnico para entrar na equipe Telecine Play ( passei ), ta aberto em meus repositórios, caso queira ver.

Abraços.

Espero ter conseguido ajudar.

**Comment by alexgwolff** [comment]
"Ainda sim não curto ter que baixar diversos runtimes de frameworks/libs pra um único site."
Micro frontend não é sinônimo de múltiplos framework.

**Comment by vicainelli** [comment]
Há alguns meses eu venho me aprofundando mais sobre esse lance.
Já vi alguns cases bacanas, como a [DAZN](https://www.youtube.com/watch?v=BuRB3djraeM) e o [Zalando](https://www.mosaic9.org/).

Semanas atrás, num meetup aqui em SP, o @gabrielrcouto falou sobre como eles fazem isso na MeMed, achei bem bacana, tem os [slides aqui](https://www.slideshare.net/gabrielrcouto/micro-frontend-cqrses-jssp-jul2019), e o [repositório](https://github.com/MemedDev/mdhub) também.

Aqui na Vivo estamos começando a nos organizar pra partir para micro frontends

**Comment by karlprieb** [comment]
Opa @juliovedovatto :)

Tenho trabalhado em um projeto com micro frontends fazem uns 6 meses e minha equipe ajudou com insights para a construção deste post :)

Estou gostando muito de trabalhar/aprender este approach e sinto que abriu muito minha cabeça para pensar de formas diferentes.

O livro recomentado pelo @Juuhz é muito bom mesmo e tras muita informação sobre como delegar responsabilidade para cada parte do código.

Se quiser conversar mais me chama :)

**Comment by brunosantanarj** [comment]
Onde eu trabalho, existe uma estratégia voltada aos micro-frontends e já temos alguns em produção. 

Mas, na minha opinião, alguns pontos precisam de maturidade, como a comunicação entre os micros. Não sei se a estratégia de postmessage(a que eu conheço e mais indicada pelos artigos) é uma boa estratégia de comunicação.

**Comment by yuritoledo** [comment]
Talvez usar o Redux (por ser "agnóstico") não seria uma boa ?

**Comment by alexsandro-xpt** [comment]
Saiu um vídeo sobre o assunto em https://github.com/frontendbr/forum/issues/661#issuecomment-525908213
